src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -253,47 +247,23 @@ async def check_pending_session_limit(
     result = True
     failure_msgs = []
 
-    kernel_query = (
-        sa.select([kernels.c.occupied_slots])
-        .select_from(kernels)
-        .where(
-            (kernels.c.status == KernelStatus.PENDING)
-            & (kernels.c.cluster_role == DEFAULT_ROLE)
-            & (kernels.c.access_key == sess_ctx.access_key)
-        )
+    pending_sessions = await SessionRow.get_pending_sessions_to_check_predicate(
+        db_sess, sess_ctx.access_key
     )
-    pending_kernels: list[dict[str, Any]] = (await db_sess.execute(kernel_query)).fetchall()
+    policy = await KeyPairResourcePolicyRow.get_pending_session_policy(db_sess, sess_ctx.access_key)
 
-    j = sa.join(
-        keypair_resource_policies,
-        keypairs,
-        keypair_resource_policies.c.name == keypairs.c.resource_policy,
-    )
-    policy_query = (
-        sa.select(
-            [
-                keypair_resource_policies.c.max_pending_session_count,
-                keypair_resource_policies.c.max_pending_session_resource_slots,
-            ]
-        )
-        .select_from(j)
-        .where(keypairs.c.access_key == sess_ctx.access_key)
-    )
-    policy: dict[str, Any] = (await db_sess.execute(policy_query)).first()
-
-    pending_count_limit: Optional[int] = policy["max_pending_session_count"]
+    pending_count_limit: int | None = policy.max_pending_session_count
     if pending_count_limit is not None and pending_count_limit > 0:
```

Comment:
Hm.. What if the pending count limit is zero but not null?
The current implementation treats "zero" as "infinity".
But this may be confusing because there is other way than reading the code to notice if "zero" means "forbidden" or "unlimited". (See lablup/backend.ai-webui#1328)

Since it is a nullable field, I think it would be clearer to treat "null" as "unlimited/unspecified".

How about your thoughts?

Reply:
I agree with you that "zero" should be different with "null".
users want to set the value of limitation as "zero".